src/mrsimulator/signal_processing/apodization.py

```diff
@@ -145,3 +177,191 @@ def str_to_quantity(cls, v, values):
     def fn(self, x):
         x = self.get_coordinates_in_units(x, unit=1.0 / self.property_units["FWHM"])
         return 1.0 if self.FWHM == 0.0 else np.exp(-self.FWHM * np.pi * np.abs(x))
+
+
+class SkewedGaussian(Apodization):
+    r"""Apodize dependent variable objects of the CSDM data with a skewed Gaussian function.
+
+    The apodization function is derived from the skewed Gaussian distribution
+
+    .. math::
+        f(x) = 2*\phi(x)*\Phi(\alpha x),
+
+    where :math:`x` are the coordinates of the dimension, and :math:`\phi` is the
+    standard normal probability density function, :math:`\Phi` is the cumulative
+    distribution function, and :math:`\alpha` is a skewing parameter. The apodization
+    function is the fourier transform of the above function which gives another
+    skewed Gaussian function and is given by
+
+    .. math::
+        f(x) = e^{-2 * (\pi * x)**2}*{1 + i*Erfi(skew*x/\sqrt(2))},
+
+    where skew is given by
+
+    .. math::
+        skew = \alpha/\sqrt(1+\alpha**2)
+
+    See https://en.wikipedia.org/wiki/Skew_normal_distribution
+
+    Arguments
+    ---------
+
+    sigma:
+        The full width at half maximum, FWHM, of the reciprocal domain Gaussian
+        function given as a string with a value and a unit. The default value is 0.
+
+    skew:
+        The skewness defining the asymmetry of the Gaussian distribution
+
+    dim_index:
+        The index of the CSDM dimension along which the operation is applied. The
+        default is the dimension at index 0.
+
+    dv_index:
+        The index of the CSDM dependent variable, where the operation is applied. If
+        not provided, the operation will be applied to every dependent variable.
+
+    Example
+    -------
+
+    >>> operation6 = sp.apodization.SkewedGaussian(skew=2, dim_index=0, dv_index=0)
+    """
+    skew: Union[float, str] = 0
+    FWHM: Union[float, str] = "2.354820045030949 Hz"
+    property_units: Dict = {"FWHM": CONST}
+
+    @validator("FWHM")
+    def str_to_quantity(cls, v, values):
+        return _str_to_quantity(v, values, "FWHM")
+
+    def fn(self, x):
+        sigma = self.FWHM / 2.354820045030949
+        x = self.get_coordinates_in_units(x, unit=1.0 / self.property_units["FWHM"])
+        prob_funct = [np.exp(-(0.5) * (sigma * j) ** 2) for j in x]
+        cumprob_funct = [0.5 + 0.5 * erf(self.skew * j / np.sqrt(2)) for j in x]
+        sg = np.asarray([a * b for a, b in zip(prob_funct, cumprob_funct)])
+        return 1.0 if self.skew == 0.0 else sg
+
+
+class Step(Apodization):
+    r"""Apodize a dependent variable of the CSDM object by a step function.
+
+    The apodization function follows
+
+    .. math::
+        f(x) = 1 if rising_edge <= x <= falling_edge ,
+        else f(x) = 0
```

Comment:
Again, it is a latex equation. All texts should be wrapped in \text{}

Reply:
There are a lot of math rendering errors. See the final output and make corrections.